src/hotspot/os_cpu/linux_riscv/vm_version_linux_riscv.cpp

```diff
@@ -135,3 +97,56 @@ void VM_Version::get_os_cpu_info() {
     fclose(f);
   }
 }
+
+void VM_Version::get_isa() {
+  char isa_buf[500];
+  strcpy(isa_buf, _isa);
+  char* saved_ptr;
+  char* isa_ext = strtok_r(isa_buf, "_", &saved_ptr);
+  while (isa_ext != nullptr) {
+    // special case for rv64* substring
+    if (!strncmp(isa_ext, "rv64", sizeof "rv64" - 1)) {
+      const char* base_ext = os::strdup(isa_ext + 4); // skip "rv64"
+      int i = 0;
+      while (base_ext[i] != '\0') {
+        const char ch = base_ext[i++];
+        if (ch == 'i') {
```

Comment:
Nit: You can use a `switch` case here, easier to read IMO.

Reply:
switch/case looks better, fixed